File: OkayegTeaTimeCSharp/Git/ReadMeGenerator.cs
using OkayegTeaTimeCSharp.JsonData;
using OkayegTeaTimeCSharp.Properties;
using System.Collections.Generic;
using System.IO;
using System.Linq;

namespace OkayegTeaTimeCSharp.Git
{
    public static class ReadMeGenerator
    {
        private static readonly string _htmlPath = Resources.ReadMePath;
        private static readonly string _title = "OkayegTeaTime";
        private static readonly string _header1 = "Commands";
        private static readonly string _header2 = "AFK-Commands";
        private static readonly List<string> _header1Text = new()
        {
            "If your channel has a custom prefix set, commands will have to start with the prefix.",
            "If your channel has no prefix set, commands will have to end with \"eg\", for example: \"pingeg\".",
            "Text in \"[ ]\" is a variable parameter"
        };
        private static readonly List<string> _cmdTableHeader = new()
        {
            "Command",
            "Alias",
            "Description [Parameter | Output]"
        };
        private static readonly List<string> _afkCmdTableHeader = new()
        {
            "Command",
            "Alias",
            "Parameter",
            "Description"
        };

        public static void GenerateReadMe()
        {
            File.WriteAllText(_htmlPath, GenerateString());
        }

        private static string GenerateString()
        {
            string result = string.Empty;
            result += Header(1, _title);
            result += Header(2, _header1);
            _header1Text.ForEach(str =>
            {
                result += str + LineBreak();
            });
            result += LineBreak();
            result += "<table><tr>";
            _cmdTableHeader.ForEach(str =>
            {
                result += TableHeader(str);
            });
            result += "</tr>";
            JsonHelper.BotData.CommandLists.Commands.OrderBy(cmd => cmd.CommandName).ToList().ForEach(cmd =>
            {
                result += $"<tr><td>{cmd.CommandName}</td><td><table>";
                cmd.Alias.ForEach(alias =>
                {
                    result += $"<tr><td>{alias}</td></tr>";
                });
                result += "</table></td><td><table>";
                for (int i = 0; i <= cmd.Parameter.Count - 1; i++)
                {
                    result += $"<tr><td>{cmd.Parameter[i]}</td><td>{cmd.Description[i]}</td></tr>";
                }
                result += "</table></td></tr>";
            });
            result += "</table>";
            result += Header(2, _header2);
            result += "<table><tr>";
            _afkCmdTableHeader.ForEach(str =>
            {
                result += $"<th>{str}</th>";
            });
            result += "</tr>";
            JsonHelper.BotData.CommandLists.AfkCommands.OrderBy(cmd => cmd.CommandName).ToList().ForEach(cmd =>
            {
                result += $"<tr><td>{cmd.CommandName}</td><td><table>";
                cmd.Alias.OrderBy(alias => alias).ToList().ForEach(alias =>
                {
                    result += $"<tr><td>{alias}</td></tr>";
                });
                result += "</table></td>";
                for (int i = 0; i <= cmd.Parameter.Count - 1; i++)
                {
                    result += $"<td>{cmd.Parameter[i]}</td><td>{cmd.Description[i]}</td>";
                }
                result += "</tr>";
            });
            result += "</table>";
            return result;
        }

        private static string Header(int i, string text)
        {
            return $"<h{i}>{text}</h{i}>";
        }

        private static string LineBreak()
        {
            return "<br />";
        }

        private static string TableHeader(string content)
        {
            return $"<th>{content}</th>";
        }
    }
}

File: OkayegTeaTimeCSharp/Tools/GitHub/ReadMeGenerator.js
const file = require("fs");
const jsonSerializer = require("../../../../JsonSerializerJS/JsonSerializer");

// Data
const commands = jsonSerializer.deserialize("../../Resources/Commands.json");
const readMePath = "../../../README.md";
const title = "OkayegTeaTime";
const header1 = "Commands";
const header2 = "AFK-Commands";
const header1Text = [
    "If your channel has a custom prefix set, commands will have to start with the prefix.",
    "If your channel has no prefix set, commands will have to end with \"eg\", for example: \"pingeg\".",
    "Text in \"[ ]\" is a variable parameter"
];
const cmdTableHeader = [
    "Command",
    "Alias",
    "Description [Parameter | Output]"
];
const afkCmdTableHeader = [
    "Command",
    "Alias",
    "Parameter",
    "Description"
];
const linebreak = "<br />";
// Data

main();

function main() {
    generateReadMe();
}

function generateReadMe() {
    try {
        file.writeFileSync(readMePath, generateString());
        console.log("Generated README.md");
    } catch (err) {
        console.error(err);
    }
}

function generateString() {
    var result = "";
    result += `<h1>${title}</h1><h2>${header1}</h2>`;
    header1Text.forEach(h => result += h + linebreak);
    result += `${linebreak}<table><tr>`;
    cmdTableHeader.forEach(c => result += `<th>${c}</th>`);
    result += "</tr>";
    commands.Commands.sort((c1, c2) => sortCmd(c1, c2)).forEach(cmd => {
        result += `<tr><td>${cmd.CommandName}</td><td><table>`;
        cmd.Alias.forEach(a => result += `<tr><td>${a}</td></tr>`);
        result += "</table></td><td><table>";
        for (var i = 0; i <= cmd.Parameter.length - 1; i++) {
            result += `<tr><td>${cmd.Parameter[i]}</td><td>${cmd.Description[i]}</td></tr>`;
        }
        result += "</table></td></tr>";
    });
    result += `</table><h2>${header2}</h2><table><tr>`;
    afkCmdTableHeader.sort((c1, c2) => sortCmd(c1, c2)).forEach(a => result += `<th>${a}</th>`);
    result += "</tr>";
    commands.AfkCommands.forEach(cmd => {
        result += `<tr><td>${cmd.CommandName}</td><td><table>`;
        cmd.Alias.forEach(a => result += `<tr><td>${a}</td></tr>`);
        result += "</table></td>";
        for (var i = 0; i <= cmd.Parameter.length - 1; i++) {
            result += `<td>${cmd.Parameter[i]}</td><td>${cmd.Description[i]}</td>`;
        }
        result += "</td></tr>";
    });
    result += "</table>";
    return result;
}

function sortCmd(cmdOne, cmdTwo) {
    if (cmdOne.CommandName < cmdTwo.CommandName) {
        return -1;
    } else if (cmdOne.CommandName < cmdTwo.CommandName) {
        return 1;
    } else {
        return 0;
    }
}
